common/network/interface.go

```diff
@@ -0,0 +1,29 @@
+package network
+
+import (
+	"fmt"
+	"net"
+)
+
+// GetFirstAddressOf returns the first address of the supplied interface name.
+func GetFirstAddressOf(name string) (string, error) {
```

Comment:
This is only ever used on the docker bridge or on the weave interface.  But will add a check.